### PR TITLE
EVG-7826 basic code gen functionality

### PR DESCRIPTION
**Pull request by john-m-liu**
There's still a lot that this doesn't handle yet, but I wanted to have this be in easily reviewable chunks. This takes a graphQL schema and generates go structs from the GQL types

### CODE REVIEW

File: rest/model/codegen.go [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+	if err != nil {
+		return nil, errors.Wrap(err, "error loading struct template")
+	}
+	fieldTemplate, err := getFieldTmpl()
+	if err != nil {
+		return nil, errors.Wrap(err, "error loading field template")
+	}
+	parsedAst, parseErr := gqlparser.LoadSchema(&source)
+	if parseErr != nil {
+		return nil, errors.New(parseErr.Error())
```
Comment (bsamek) [inline rest/model/codegen.go:59]: I would expect `Error()` to do the same thing as string formatting, so you can just `errors.Wrap(err, "helpful text")` (and rename `parseErr` to `err`).

File: rest/model/codegen.go [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+		typeNames = append(typeNames, key)
+	}
+	sort.Strings(typeNames)
+	for _, typeName := range typeNames {
+		gqlType := parsedAst.Types[typeName]
+		if gqlType.BuiltIn {
+			continue
+		}
+		if gqlType.Kind == ast.Scalar {
+			continue // TODO: add more validation here
```
Comment (bsamek) [inline rest/model/codegen.go:75]: ping on TODO

File: rest/model/codegen.go (hunk 1 above)
Comment (john-m-liu) [reply rest/model/codegen.go:59]: Oh it was a different type but I didn't realize it implemented the error interface. Fixed

File: rest/model/codegen.go (hunk 2 above)
Comment (john-m-liu) [reply rest/model/codegen.go:75]: This is coming in later parts, removed the todo

**Pull request merged**